### PR TITLE
Fixes zombie infection holders

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Automatically deletes them if their host mob is destroyed. Thought gibbing would already do that, sorry. Makes them invisible as well just in case.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 1]
```diff
@@ -146,12 +146,22 @@
...
 	desc = "You shouldn't be seeing this."
+	invisibility = INVISIBILITY_OBSERVER
 	unsuitable_atmos_damage = 0
 	stat_attack = 2
 	gold_core_spawnable = 0
 	AIStatus = AI_OFF
 	stop_automated_movement = 1
+	density = 0
+	
+/mob/living/simple_animal/hostile/zombie/holder/Life()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: don't call this every ```Life()```
Put it in ```Move()```, check that the newloc is not a human, and then delete them.

**Comment by Razharas** [comment]
@KorPhaeron do the remie

File: code/modules/mob/living/simple_animal/hostile/zombie.dm [hunk 2]
```diff
@@ -146,12 +146,22 @@
 
 /mob/living/simple_animal/hostile/zombie/holder
 	name = "infection holder"
+	icon_state = "none"
+	icon_living = "none"
+	icon_dead = "none"
 	desc = "You shouldn't be seeing this."
+	invisibility = INVISIBILITY_OBSERVER
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/zombie.dm]: Do the ghost really need to be able to see this? why not invisibility = 101 ?

**Comment by duncathan** [comment]
@KorPhaeron fix Phil and remie's concerns

**Comment by KorPhaeron** [comment]
@duncathan 

I just made them fully invisible/non dense, they'll delete themselves from their New() code already anyway and nobody will be able to see or interact with them in the meantime.

Move() wont work since its just dumping them in the tile

**Comment by Dorsisdwarf** [comment]
so is it freddy?

**Comment by KorPhaeron** [comment]
@Dorsisdwarf 
Yes

**Comment by Razharas** [comment]
I see no complaints left open
Looks like some kind of fix to me
Merged

**Pull request merged**